src/test/java/seedu/address/logic/commands/internships/role/DeleteInternshipRoleCommandTest.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.logic.commands.internships.role;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandAssignmentTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_IN_LIST;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_IN_LIST;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.CommandAssignmentTestUtil;
+import seedu.address.logic.commands.internship.role.DeleteInternshipRoleCommand;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.internship.role.InternshipRole;
+import seedu.address.testutil.TypicalInternshipRoles;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code DeleteInternshipRoleCommand}.
+ */
+
+public class DeleteInternshipRoleCommandTest {
+
+    private Model model = new ModelManager(TypicalInternshipRoles.getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        InternshipRole internshipRoleToDelete =
+                model.getFilteredInternshipRoleList().get(INDEX_FIRST_IN_LIST.getZeroBased());
+        DeleteInternshipRoleCommand deleteCommand = new DeleteInternshipRoleCommand(INDEX_FIRST_IN_LIST);
+
+        String expectedMessage = String.format(DeleteInternshipRoleCommand.MESSAGE_DELETE_INTERNSHIP_ROLE_SUCCESS,
+                Messages.format(internshipRoleToDelete));
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.deleteInternshipRole(internshipRoleToDelete);
+        assertCommandSuccess(deleteCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_validIndexFilteredList_success() {
```

Comment:
For this I think it would be better if you `updateFilteredInternshipRoleList` with a predicate first before executing then check if the correct one was deleted.